### PR TITLE
MCKIN-12934 Upgraded to django 2.x

### PR DESCRIPTION
**Pull request by nasirhjafri**

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @nasirhjafri! I've created [OSPR-4097](https://openedx.atlassian.net/browse/OSPR-4097) to keep track of it in JIRA. JIRA is a place for product owners to prioritize feature reviews by the engineering development teams.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- edx-code email threads
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will still be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).



We can't start reviewing your pull request until you've submitted a     [signed contributor agreement](https://open.edx.org/wp-content/uploads/2019/01/individual-contributor-agreement.pdf)     or indicated your institutional affiliation. Please see the [CONTRIBUTING](https://github.com/edx/edx-platform/blob/master/CONTRIBUTING.rst)     file for more information.

File: testserver/urls.py [hunk 1]
```diff
@@ -4,19 +4,20 @@
...
+    path('', testserver_views.index),
+    path('accounts/login/', LoginView.as_view(template_name='registration/login.html')),
+    path('logout/', testserver_views.logout_page),
+    path('register/', testserver_views.register),
+    path('register/success/', testserver_views.register_success),
 
-    url(r'^api/', include('edx_notifications.server.api.urls')),
-    url(r'^jsi18n/$', JavaScriptCatalog.as_view(), name='javascript-catalog'),
+    path('api/', include('edx_notifications.server.api.urls')),
+    path('jsi18n/', JavaScriptCatalog.as_view(), name='javascript-catalog'),
```
Comment (ihtram) [inline testserver/urls.py]: I need a little context about removing regex here. we could have used re_path with existing regex.

File: package.json [hunk 2]
```diff
@@ -15,7 +15,7 @@
     "karma-jasmine": "latest",
     "karma-jasmine-jquery": "latest",
     "karma-phantomjs-launcher": "latest",
-    "sinon": "latest",
+    "sinon": "<=7.5.0",
```
[the remark this answers is not in the document]
Comment (nasirhjafri) [reply package.json]: `sinon 8` was released few weeks back and have some major changes which are causing the tests to break.

File: setup.py [hunk 3]
```diff
@@ -37,7 +38,7 @@ def load_requirements(*requirements_paths):
 
 setup(
     name='edx-notifications',
-    version='1.0.2',
+    version='1.0.3',
```
[the remark this answers is not in the document]
Comment (nasirhjafri) [reply setup.py]: sure!

File: testserver/urls.py (hunk 1 above)
Comment (nasirhjafri) [reply testserver/urls.py]: path() always matches the complete path, so path('jsi18n/') is equivalent to url('^jsi18n/$')
So no need to use regexes over here.

**Comment by nasirhjafri** [comment]
@ihtram Please review it again, thanks!

**Comment by ihtram** [comment]
@nasirhjafri Looks good now. Can you please squash the commits and push a single commit?

**Comment by nasirhjafri** [comment]
@ihtram Squashed the commit to a single commit.

**Pull request merged**